From 3533608c930492548cbc89417614098e6896fa72 Mon Sep 17 00:00:00 2001
From: Jeremie GARCIA <jeremie.garcia@intel.com>
Date: Thu, 26 Sep 2013 12:47:31 +0200
Subject: [PDK KK AOSP]wifi: add a NO_PRIVATE_LIB define

BZ: 141359

Prevent "driver_nl80211.c" from refereing to an external
"wpa_driver_nl80211_driver_cmd" function when the vendor
used doesn't come with a private lib.

Change-Id: I77d77ea39ec7017e6fbf358f126fa276c3fe2803
Category: device enablement
Domain: CWS.WIFI-TI
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
---
 src/drivers/driver_nl80211.c |    2 +-
 wpa_supplicant/Android.mk    |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index a3ff189..c61740e 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -10837,7 +10837,7 @@ const struct wpa_driver_ops wpa_driver_nl80211_ops = {
 	.get_noa = wpa_driver_get_p2p_noa,
 	.set_ap_wps_ie = wpa_driver_set_ap_wps_p2p_ie,
 #endif
-#ifdef ANDROID
+#if defined(ANDROID) && !defined(NO_PRIVATE_LIB)
 	.driver_cmd = wpa_driver_nl80211_driver_cmd,
 #endif
 };
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 8515b5a..350ec96 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -1550,6 +1550,8 @@ LOCAL_STATIC_LIBRARIES := libCustomWifi
 endif
 ifneq ($(BOARD_WPA_SUPPLICANT_PRIVATE_LIB),)
 LOCAL_STATIC_LIBRARIES += $(BOARD_WPA_SUPPLICANT_PRIVATE_LIB)
+else
+L_CFLAGS += -DNO_PRIVATE_LIB
 endif
 LOCAL_SHARED_LIBRARIES := libc libcutils liblog
 ifeq ($(CONFIG_TLS), openssl)
-- 
1.7.4.1

